git.c

```diff
@@ -318,6 +324,13 @@ static int handle_options(const char ***argv, int *argc, int *envchanged)
 		(*argv)++;
 		(*argc)--;
 	}
+	if (advice_set_git_dir_without_worktree && git_dir_is_set &&
+	    !work_tree_is_set)
```

Comment:
I think this also needs an `&& !is_bare_repository()`, both to silence the test failure and to be correct.

Reply:
Thank you, let me give this a try